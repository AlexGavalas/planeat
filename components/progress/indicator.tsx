import { Box, Title } from '@mantine/core';
import { NavArrowDown } from 'iconoir-react';

import styles from './indicator.module.css';

type Section = {
    label: string;
    percent: number;
    bg: string;
};

interface ProgressIndicatorProps {
    label: string;
    value: number;
    percent: number;
    sections: Section[];
}

export const ProgressIndicator = ({
    label,
    percent,
    value,
    sections,
}: ProgressIndicatorProps) => {
    return (
        <Box>
            <Title order={3}>{label}</Title>
            <div
                className={styles.indicator}
                style={{
                    left: `calc(${percent}% - 50px)`,
                }}
            >
                {value} %
                <NavArrowDown />
            </div>
            <div className={styles.sections}>
                {sections.map(({ label, percent, bg }) => (
                    <div
                        key={label}
                        className={styles.section}
                        style={{
                            width: `${percent}%`,
                            backgroundColor: bg,
                        }}
                    >
                        {label}
                    </div>
                ))}
            </div>
        </Box>
    );
};
