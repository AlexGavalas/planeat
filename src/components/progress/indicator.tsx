import { Progress, Stack, Text, Title } from '@mantine/core';
import { NavArrowDown } from 'iconoir-react';

import { type Section } from '~types/types';

import styles from './indicator.module.css';

interface ProgressIndicatorProps {
    label: string;
    value: number | null;
    percent: number | null;
    sections: (Section & { label: string })[];
}

export const ProgressIndicator = ({
    label,
    percent,
    value,
    sections,
}: ProgressIndicatorProps) => {
    return (
        <Stack gap="xs">
            <Title order={3}>{label}</Title>
            {Boolean(value) && (
                <div
                    className={styles.indicator}
                    style={{
                        left: `calc(${percent}% - 50px)`,
                    }}
                >
                    <Text>{value}%</Text>
                    <NavArrowDown />
                </div>
            )}
            <Progress.Root
                size="xl"
                style={{
                    height: '2rem',
                }}
            >
                {sections.map((section) => (
                    <Progress.Section
                        key={section.key}
                        value={section.percent}
                        color={section.bg}
                    >
                        <Progress.Label>{section.label}</Progress.Label>
                    </Progress.Section>
                ))}
            </Progress.Root>
        </Stack>
    );
};
