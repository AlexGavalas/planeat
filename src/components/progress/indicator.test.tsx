import { renderWithUser } from '~test/utils';

import { ProgressIndicator } from './indicator';

describe('<ProgressIndicator />', () => {
    it('renders the sections', () => {
        const { container } = renderWithUser(
            <ProgressIndicator
                label="Test"
                percent={50}
                sections={[
                    {
                        key: 'test',
                        label: 'Test',
                        percent: 50,
                        bg: 'blue',
                    },
                ]}
                value={50}
            />,
        );

        expect(container).toMatchSnapshot();
    });
});
