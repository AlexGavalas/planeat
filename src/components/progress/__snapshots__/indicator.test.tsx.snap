// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressIndicator /> renders the sections 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-xs);"
  >
    <h3
      class="m-8a5d1357 mantine-Title-root"
      data-order="3"
      style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size);"
    >
      Test
    </h3>
    <div
      class="indicator"
      style="left: calc(50% - 50px);"
    >
      <p
        class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
      >
        50
        %
      </p>
      <svg
        color="currentColor"
        fill="none"
        height="1.5em"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="1.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 9l6 6 6-6"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
    <div
      class="m-db6d6462 mantine-Progress-root"
      data-size="xl"
      style="--progress-size: var(--progress-size-xl); height: 2rem;"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        aria-valuetext="50%"
        class="m-2242eb65 mantine-Progress-section"
        role="progressbar"
        style="--progress-section-width: 50%; --progress-section-color: var(--mantine-color-blue-filled);"
      >
        <div
          class="m-91e40b74 mantine-Progress-label"
        >
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;
